### DISCUSSION TITLE
Dark Sky weather service replacement

### DISCUSSION THREAD

**Discussion by ppac** [alex-hhh/ActivityLog2#93] [Ideas]
The automatic download of data from Dark Sky has been removed in the latest version, but fortunately I found an alternative today that tries to do exactly the same thing as Dark Sky (similar API-respone): https://pirateweather.net

If it fits the needs of ActivityLog2, you might want to check it out and implement it.

**Comment by alex-hhh** [comment]
If you haven't already seen it, there is an issue about weather data which contains some notes.  In particular, it lists some requirements for the weather data API.  An important requirement is to have historical weather, that is to fetch the weather conditions from a few hours to a few days ago.  Looks like pirate weather does not have that feature (yet?):

https://github.com/alex-hhh/ActivityLog2/issues/46

Second, I have a separate Python script that fetches weather data from the Meteostat service and updates the ActivityLog2 database directly.  You can use it either as is (you'll have to point it to your database), or as an example on how to write a "Weather data updater" from any weather service you prefer.  I understand that this might not be convenient for everyone, especially for non-technical people.

https://gist.github.com/alex-hhh/e1a4ffe3e861ad3fd31f6655a3a66b26

I might revisit this in the future, but for now, I have limited time to work on this project, and that time is used to work on other features.
